Lib/selectors.py

```diff
@@ -452,23 +455,28 @@ def select(self, timeout=None):
             # epoll_wait() expects `maxevents` to be greater than zero;
             # we want to make sure that `select()` can be called when no
             # FD is registered.
-            max_ev = max(len(self._fd_to_key), 1)
+            max_ev = len(self._fd_to_key) or 1
 
             ready = []
             try:
                 fd_event_list = self._selector.poll(timeout, max_ev)
             except InterruptedError:
                 return ready
-            for fd, event in fd_event_list:
-                events = 0
-                if event & ~select.EPOLLIN:
-                    events |= EVENT_WRITE
-                if event & ~select.EPOLLOUT:
-                    events |= EVENT_READ
 
-                key = self._fd_to_key.get(fd)
+            fd_to_key = self._fd_to_key
+            for fd, event in fd_event_list:
+                key = fd_to_key.get(fd)
                 if key:
-                    ready.append((key, events & key.events))
+                    ready.append(
+                        (
+                            key,
+                            (
+                                (event & _NOT_EPOLLIN and EVENT_WRITE)
+                                | (event & _NOT_EPOLLOUT and EVENT_READ)
+                            )
+                            & key.events
+                        )
+                    )
```

Comment:
I feel this is going too far.
How about splitting this into two lines?

```suggestion
                    events = ((event & _NOT_EPOLLIN and EVENT_WRITE)
                              | (event & _NOT_EPOLLOUT and EVENT_READ))
                    ready.append((key, events & key.events))
```

Reply:
That looks nicer 👍

Reply:
thanks